### ISSUE TITLE
WebSocket Error Handling in WebAssembly

### ISSUE THREAD

**Issue by jaydo25** [jgraef/reqwest-websocket#16]
I am trying to use a websocket and test scenerios where it can't get through the initial upgrade request using WASM-32. Upon a failure, the error in the compiled javascript code is not being passed into the Rust code properly, and I think it is due to this [line](https://github.com/jgraef/reqwest-websocket/blob/eb42bdf27efc0471d7890c0e7a520626e34f68d4/src/wasm.rs#L34), where there is assumed to always be an error message. In my case, there is no error message, and this causes the entire program to crash. Here is my stack trace:

```
internal-client-b79b8b5eb88fa12f.js:65 Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at passStringToWasm0 (internal-client-b79b8b5eb88fa12f.js:65:19)
    at imports.wbg.__wbg_message_730094b985f3ba2e (internal-client-b79b8b5eb88fa12f.js:1202:18)
    at internal_client-9db5fb7e5c142151.wasm.web_sys::features::gen_ErrorEvent::ErrorEvent::message::hb3ba2710fd13fa64 (internal-client-b79b8b5eb88fa12f_bg.wasm:0xcd48b1)
    at internal_client-9db5fb7e5c142151.wasm.<reqwest_websocket::wasm::WebSysError as core::convert::From<web_sys::features::gen_ErrorEvent::ErrorEvent>>::from::h57d2181758beda77 (internal-client-b79b8b5eb88fa12f_bg.wasm:0xcbfeb9)
    at internal_client-9db5fb7e5c142151.wasm.<T as core::convert::Into<U>>::into::h696748b264db0539 (internal-client-b79b8b5eb88fa12f_bg.wasm:0xe21857)
    at internal_client-9db5fb7e5c142151.wasm.reqwest_websocket::wasm::WebSysWebSocketStream::new::{{closure}}::{{closure}}::ha4ca86e6f342a56a (internal-client-b79b8b5eb88fa12f_bg.wasm:0x6d1b5d)
    at internal_client-9db5fb7e5c142151.wasm.<dyn core::ops::function::FnMut<(:9443/A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h2e6c261e3cd29667 (https://localhost:9443/asset-server/assets/find/Credera/United%20States/app/internal-client-b79b8b5eb88fa12f_bg.wasm)
    at __wbg_adapter_41 (internal-client-b79b8b5eb88fa12f.js:228:10)
    at WebSocket.real (internal-client-b79b8b5eb88fa12f.js:209:20)
passStringToWasm0 @ internal-client-b79b8b5eb88fa12f.js:65
imports.wbg.__wbg_message_730094b985f3ba2e @ internal-client-b79b8b5eb88fa12f.js:1202
$web_sys::features::gen_ErrorEvent::ErrorEvent::message::hb3ba2710fd13fa64 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xcd48b1
$<reqwest_websocket::wasm::WebSysError as core::convert::From<web_sys::features::gen_ErrorEvent::ErrorEvent>>::from::h57d2181758beda77 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xcbfeb9
$<T as core::convert::Into<U>>::into::h696748b264db0539 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xe21857
$reqwest_websocket::wasm::WebSysWebSocketStream::new::{{closure}}::{{closure}}::ha4ca86e6f342a56a @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x6d1b5d
$<dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h2e6c261e3cd29667 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xc2dd23
__wbg_adapter_41 @ internal-client-b79b8b5eb88fa12f.js:228
real @ internal-client-b79b8b5eb88fa12f.js:209
error (async)
imports.wbg.__wbg_setonerror_39a785302b0cd2e9 @ internal-client-b79b8b5eb88fa12f.js:804
$web_sys::features::gen_WebSocket::WebSocket::set_onerror::h5307893996413682 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xd62801
$reqwest_websocket::wasm::WebSysWebSocketStream::new::{{closure}}::hcae057a6ec632018 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x37047c
$reqwest_websocket::UpgradedRequestBuilder::send::{{closure}}::h590c0b70450cbbb6 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x6d46d4
$reqwest_websocket::websocket::{{closure}}::ha1844510cedfbd36 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x6b2fe0
$internal_client::data::http_client::websocket_browser::{{closure}}::hebc6c48c9423058c @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x5a2275
$internal_client::data::http_client::HttpClient::websocket_send_write::{{closure}}::h22e857b744b44de0 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x74c456
$internal_client::domain::chat::submit_message::submit_message::{{closure}}::h299d4161f6a48fa7 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x2b17f1
$internal_client::ui::util::input_form::input_form::__InputForm::{{closure}}::{{closure}}::hc4e8599d95b4c881 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x8b289f
$wasm_bindgen_futures::task::singlethread::Task::run::h016511925a7290ab @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x9c36e3
$wasm_bindgen_futures::queue::QueueState::run_all::h896e56b7ec4e336f @ internal-client-b79b8b5eb88fa12f_bg.wasm:0x968452
$wasm_bindgen_futures::queue::Queue::new::{{closure}}::h49028293b6b33a54 @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xd37168
$<dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h11d27d6aff35c9bd @ internal-client-b79b8b5eb88fa12f_bg.wasm:0xc310db
__wbg_adapter_68 @ internal-client-b79b8b5eb88fa12f.js:264
real @ internal-client-b79b8b5eb88fa12f.js:209
```

**Comment by jgraef** [comment]
Thanks for the report!

The issue is that we assumed the `onerror` callback gets called with an `ErrorEvent`, but it's a generic `Event`[1].

I opened a PR (#18) with a fix. Can you let me know, if this fixes it?

[1]: https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/error_event

**Comment by jgraef** [comment]
The patch is released as version 0.4.1.
